x/concentrated-liquidity/lp_test.go

```diff
@@ -276,3 +278,108 @@ func mergeConfigs(dst *lpTest, overwrite *lpTest) {
 		}
 	}
 }
+
+func (s *KeeperTestSuite) TestSendCoinsBetweenPoolAndUser() {
+	type sendTest struct {
+		coin0       sdk.Coin
+		coin1       sdk.Coin
+		poolToUser  bool
+		expectError bool
+	}
+	tests := map[string]sendTest{
+		"asset0 and asset1 are positive, position creation (user to pool)": {
+			coin0: sdk.NewCoin("eth", sdk.NewInt(1000000)),
+			coin1: sdk.NewCoin("usdc", sdk.NewInt(1000000)),
+		},
+		"only asset0 is positive, position creation (user to pool)": {
+			coin0: sdk.NewCoin("eth", sdk.NewInt(1000000)),
+			coin1: sdk.NewCoin("usdc", sdk.NewInt(0)),
+		},
+		"only asset1 is positive, position creation (user to pool)": {
+			coin0: sdk.NewCoin("eth", sdk.NewInt(0)),
+			coin1: sdk.NewCoin("usdc", sdk.NewInt(1000000)),
+		},
+		"only asset0 is greater than sender has, position creation (user to pool)": {
+			coin0:       sdk.NewCoin("eth", sdk.NewInt(100000000000000)),
+			coin1:       sdk.NewCoin("usdc", sdk.NewInt(1000000)),
+			expectError: true,
+		},
+		// "only asset1 is greater than sender has, position creation (user to pool)": {
+		// 	coin0:       sdk.NewCoin("eth", sdk.NewInt(1000000)),
+		// 	coin1:       sdk.NewCoin("usdc", sdk.NewInt(100000000000000)),
+		// 	expectError: true,
+		// },
```

Comment:
for some reason, everything works as expected but this...

if the second asset is too large, we end up BankSending the coin0 but not coin1...

Reply:
Actually, I suppose this makes sense. While this test case will be odd and show a partial send, it wouldn't actually get committed and would get reverted due to the error right? Yeah I guess for error cases I should just not check the resulting amounts